lib/src/dropdown/dropdown.component.ts

```diff
@@ -0,0 +1,503 @@
+            this.nativeOnChange && this.nativeOnChange(event);
+            this.selectedValue = item;
+        } else {
+            const items = Array.from(event.target.options as HTMLOptionsCollection)
+                .filter(e => e.selected)
+                .map(e => {
+                    const item = this.list.filter((el: Item) => el.key === e.value)[0];
+                    return item;
+                });
+            this.nativeOnChange && this.nativeOnChange(event);
+            this.selectedValue = items;
+        }
+    }
+
+    /** Function which handles the logic of setting the non-native onChange prop (and sets the internal selected value as well) */
+    handleOnChange(value: Item | Array<Item>): void {
+        this.nativeOnChange && this.nativeOnChange(value);
+        this.selectedValue = value;
+    }
+
+    /** Function containing the clear button logic */
+    handleClear(): void {
+        this.handleOnChange(null);
+        this.open = false;
+    }
+
+    /** The native onchange event function that runs when the search input value changes */
+    handleOnChangeSearch(event: any): void {
+        if (this.currentFocused !== -1) {
+            this.currentFocused = -1;
+        }
+        this.searchText = event.target.value;
+    }
+
+    /** Function containing the select dropdown item logic */
+    optionItemSelected(item: Item): void {
+        if (!this.multi) {
+            const newItem = { ...item };
+            this.handleOnChange(newItem);
+            this.open = false;
+        } else {
+            const currentList = (this.selectedValue as Array<Item>) ? (this.selectedValue as Array<Item>) : [];
+            const index = currentList.findIndex((e: Item) => e.key === item.key);
+            if (index === -1) {
+                const newItem = { ...item };
+                const newList = [...currentList, newItem];
+                this.handleOnChange(newList);
+            } else {
+                const newList = currentList.filter((e: Item) => e.key !== item.key);
+                this.handleOnChange(newList);
+            }
+        }
+        this.handleFocus();
+    }
+
+    /** The native event function that runs when the dropdown button is clicked */
+    handleClickToggle(event: any): void {
+        if (!this.disabled) {
+            this.open = !this.open;
+        }
+    }
+
+    /** Function containing the select all button logic */
+    handleSelectAll(): void {
+        if (this.allSelected) {
+            this.handleOnChange([]);
+        } else {
+            this.handleOnChange(this.list);
+        }
+        this.handleFocus();
+    }
+
+    // HELPERS ================================
+    /** Returns the appropriate title for different situations and component types */
+    getTitleLabel() {
+        if (this.uniqueList && this.uniqueList.length === 0) {
+            return "Empty";
+        }
+        if (this.selectedList && this.selectedList.length > 0) {
```

Comment:
This block will not be executed if the first if is true right ? It is an `else if` then isn't it ?

Reply:
see my other comment